test/extensions.js

```diff
@@ -39,10 +39,12 @@ tape('send and receive extension messages', function (t) {
       t.equal(message.toString('hex'), EXAMPLE_MESSAGE.toString('hex'))
     })
 
-    feed1.on('peer-add', function () {
-      feed1.extension(EXAMPLE_TYPE, EXAMPLE_MESSAGE)
+    feed1.once('peer-add', function () {
+      setImmediate(function () {
```

Comment:
Why does the `extension` call need to be `setImmediate`d here?

Reply:
It's because the peer-add event is trigger as soon as the peer is added (ie feed.replicate() tick) but it's not setup yet. we prop want to add a peer-ready event for this in general